### ISSUE TITLE
Bug in _openssl's aes_cbc_no_padding_encrypt?

### ISSUE THREAD

**Issue by nathforge** [wbond/oscrypto#24]
I'm trying to call `oscrypto.symmetric.aes_cbc_no_padding_encrypt` with a 16 byte key, 16 byte IV, and 64 bytes of data.

This throws an error:

```python
>>> import oscrypto.symmetric
>>>
>>> key  = [0]   * 16
>>> iv   = [1]   * 16
>>> data = [255] * 64
>>>
>>> oscrypto.symmetric.aes_cbc_no_padding_encrypt(key=bytes(key), data=bytes(data), iv=bytes(iv))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "oscrypto/_openssl/symmetric.py", line 74, in aes_cbc_no_padding_encrypt
    return (iv, _encrypt(cipher, key, data, iv, False))
  File "oscrypto/_openssl/symmetric.py", line 606, in _encrypt
    raise ValueError('padding must be specified')
ValueError: padding must be specified
```

---

Here's the abridged source of [oscrypto._openssl.symmetric](https://github.com/wbond/oscrypto/blob/master/oscrypto/_openssl/symmetric.py):

```
def aes_cbc_no_padding_encrypt(key, data, iv):
    cipher = _calculate_aes_cipher(key)
    return (iv, _encrypt(cipher, key, data, iv, False))

def _calculate_aes_cipher(key):
    if len(key) == 16:
        return 'aes128'

def _encrypt(cipher, key, data, iv, padding):
    if cipher != 'rc4' and not padding:
        raise ValueError('padding must be specified')
    # ...
```

As I read it, `aes_cbc_no_padding_encrypt` always passes `cipher != rc4` and `padding = False` to `_encrypt`, resulting in the exception.

---

Happy to help with a fix, just wanted to check I'm not doing anything crazy?

**Comment by wbond** [comment]
Sorry that I didn't notice this issue for so long. I'm hoping to have a few minutes to look into it over the weekend.

**Comment by alanlonglong** [comment]
how to solve it ?

**Comment by wbond** [comment]
This should be fixed by 67d0017957ef024e18cbe5522e365771ed1b0e94

**Comment by wbond** [comment]
This is part of 1.0.0 which is now available on PyPi https://pypi.org/project/oscrypto/
